Doc/library/asyncio-stack.rst

```diff
@@ -0,0 +1,139 @@
+.. currentmodule:: asyncio
+
+
+.. _asyncio-stack:
+
+===================
+Stack Introspection
+===================
+
+**Source code:** :source:`Lib/asyncio/stack.py`
+
+-------------------------------------
+
+asyncio has powerful runtime call stack introspection utilities
```

Comment:
@1st1 This is awesome! To reduce user confusion do we want to mention that this will typically be used by profiling and debug tools?

Reply:
Thank you, Carol! I've added this quick diff:
![CleanShot 2024-10-16 at 12 26 43@2x](https://github.com/user-attachments/assets/70834fa8-eabc-405e-871a-451ae883079d)